src/olympia/lib/crypto/packaged.py

```diff
@@ -172,3 +179,21 @@ def is_signed(file_path):
         filenames = set()
     return set([u'META-INF/mozilla.rsa', u'META-INF/mozilla.sf',
                 u'META-INF/manifest.mf']).issubset(filenames)
+
+
+def get_signature_ou(pkcs7):
+    """
+    Extracts the ou out of a DER formatted, detached PKCS7
+    signature buffer
+    """
+    content = cms.ContentInfo.load(pkcs7)['content'].native
+
+    # Fetch the certificate stack that is the list of signers
+    # Since there should only be one in this use case, take the zeroth
```

Comment:
This is incorrect. Every add-on we sign has two certificates: the end-entity signer, and the intermediate issuer. The third certificate, the root CA, is hardcoded in Firefox. All three are used to build a chain a trust used to validate the add-on signature.